packages/appeals-service-api/src/routes/v2/appeal-cases/repo.js

```diff
@@ -39,17 +82,142 @@ class AppealCaseRepository {
 	 * @returns {Promise<AppealCase>}
 	 */
-	putByCaseReference(caseReference, data) {
-		return this.dbClient.appealCase.upsert({
-			create: { ...data, Appeal: { create: {} } }, // create with parent Appeal
-			update: data,
+	async putHASByCaseReference(caseReference, data) {
+		const mappedData = mapHASDataModelToAppealCase(data);
+
+		// todo: move this to mapping
+		delete mappedData.caseType;
+		delete mappedData.linkedCaseStatus;
+		delete mappedData.leadCaseReference;
+		delete mappedData.notificationMethod;
+		delete mappedData.nearbyCaseReferences;
+		delete mappedData.neighbouringSiteAddresses;
+		delete mappedData.affectedListedBuildingNumbers;
+
+		const appealCase = await this.dbClient.appealCase.upsert({
+			create: { ...mappedData, Appeal: { create: {} } },
+			update: mappedData,
 			where: {
 				caseReference
 			}
 		});
+
+		// case relations
+		await this.dbClient.$transaction(async (tx) => {
+			await tx.appealCaseRelationship.deleteMany({
+				where: {
+					OR: [
+						{
+							caseReference: caseReference
+						},
+						{
+							caseReference2: caseReference
+						}
+					]
+				}
+			});
+
+			if (data.nearbyCaseReferences?.length) {
+				const direction1 = data.nearbyCaseReferences.map((nearby) => ({
+					caseReference: caseReference,
+					caseReference2: nearby
+				}));
+				const direction2 = data.nearbyCaseReferences.map((nearby) => ({
+					caseReference: nearby,
+					caseReference2: caseReference
+				}));
+
+				await tx.appealCaseRelationship.createMany({
+					data: direction1.concat(direction2)
+				});
+			}
+
+			if (data.leadCaseReference) {
+				await tx.appealCaseRelationship.create({
+					data: {
+						caseReference,
+						caseReference2: data.leadCaseReference,
+						type: 'linked' // todo: const this
+					}
+				});
+			}
+		});
+
+		// neighbour addresses
+		await this.dbClient.$transaction(async (tx) => {
+			await tx.neighbouringAddress.deleteMany({
+				where: {
+					caseReference: caseReference
+				}
+			});
+
+			if (data.neighbouringSiteAddresses?.length) {
+				await tx.neighbouringAddress.createMany({
+					data: data.neighbouringSiteAddresses.map((address) => ({
+						caseReference,
+						addressLine1: address.neighbouringSiteAddressLine1,
+						addressLine2: address.neighbouringSiteAddressLine2,
+						townCity: address.neighbouringSiteAddressTown,
+						county: address.neighbouringSiteAddressCounty,
+						postcode: address.neighbouringSiteAddressPostcode,
+						siteAccessDetails: address.neighbouringSiteAccessDetails,
+						siteSafetyDetails: address.neighbouringSiteSafetyDetails
+					}))
+				});
+			}
+		});
+
+		// listed building
+		await this.dbClient.$transaction(async (tx) => {
+			await tx.appealCaseListedBuilding.deleteMany({
+				where: {
+					caseReference: caseReference
+				}
+			});
+
+			if (data.affectedListedBuildingNumbers?.length) {
+				// todo: inefficient promise.all?
+				data.affectedListedBuildingNumbers.forEach(async (reference) => {
```

Comment:
adding listed buildings here in case they don't exist, will only have reference if they come in this way.
need to transfer our data population script/front end to use this table instead of mongo still